src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,183 @@
+                keys['properties'].append(parent_name + property[0].name)
+        if isinstance(obj, HasConditionTemplates):
+            for condition in obj.conditions:
+                keys['conditions'].append(parent_name + condition[0].name)
+        if isinstance(obj, HasParameterTemplates):
+            for parameter in obj.parameters:
+                keys['parameters'].append(parent_name + parameter[0].name)
+    return keys
+
+
+def auto_configure_material(
+        *,
+        project: Project,
+        material: Union[MaterialRun, LinkByUID, str, UUID],
+        score: Score,
+        mode: str = 'SIMPLE',
+        label: str = '',
+        print_status_info: bool = False,
+) -> DesignExecution:
+    """[ALPHA] Description.
+
+    Parameters
+    ----------
+    project: Project
+        Project to use when accessing the Citrine Platform
+    material: Union[MaterialRun, LinkByUID, str, UUID]
+        Sample material to extract descriptor keys
+    score: Score
+        Scoring function used to rank candidates during design execution.
+        Must contain objectives/constraints with matching descriptor keys
+        to those found within the provided material history.
+    mode: str
+        Option in {'SIMPLE', 'FORMULATION'} that controls formatting descriptor keys
+        Default: 'SIMPLE'
+    label: str
+        Naming label to affix to auto-configured assets on the Citrine Platform
+        Default: ''
+    print_status_info: bool
+        Whether to print the status of predictor, design space, and design workflow validation
+        Default: False
+
+    Returns
+    -------
+    DesignExecution
+        Triggered design execution given the auto-configured workflow and provided `score`
+
+    """
+    if mode not in {"FORMULATION", "SIMPLE"}:
+        msg = "Called with mode: {}.  Expected 'SIMPLE' or 'FORMULATION'.".format(mode)
+        raise ValueError(msg)
+
+    suffix = ' - {}'.format(label) if label else ''
+
+    # Mapping of input mode to the config input options (different for each case)
+    config_options = {
+        'FORMULATION': (TableBuildAlgorithm.FORMULATIONS, 'FORMULATION'),
+        'SIMPLE': (TableBuildAlgorithm.SINGLE_ROW, 'PLAIN')
+    }
+    table_algorithm, predictor_pattern = config_options[mode]
+
+    print('Building default GEM table from material history...')
+    table_config, _ = project.table_configs.default_for_material(
+        material=material, name=f'Default GEM Table{suffix}', algorithm=table_algorithm
+    )
+    table_config = project.table_configs.register(table_config)
+    table = project.tables.build_from_config(table_config)
+
+    print('Building default predictor from GEM table...')
+    formulation = None
+    if mode == 'FORMULATION':
+        formulation = FormulationDescriptor('Formulation descriptor')
+    data_source = GemTableDataSource(
+        table_id=table.uid, table_version=table.version, formulation_descriptor=formulation
```

Comment:
I believe it will work without `formulation_descriptor`, since we fill in a default one if we see formulation data but none is provided.  The descriptor key is just "Formulation", which you could get from the [`descriptors_from_data_source` method](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/resources/descriptors.py#L48)

Reply:
Tested and that is the case, will remove the conditional check